### DISCUSSION TITLE
Unity 3D visualization in Avalonia app

### DISCUSSION THREAD

**Discussion by NemanjaBozovic-TomTom** [AvaloniaUI/Avalonia#9151] [General]
Hell,

I'm interested in in integrating Unity-based rendering engine in different Ui framework and I'm wondering if Avalonia supports this?

Pardon my uninformed question, but I'm just started exploring this space and I'm understand my question might not make sense, so I'll expand further.

Basically we have map rendered created (WIP) in Unity and we want to build rest of the application (ui widgets, buttons, labels, etc) around it. We would like to use ui framework like Avalonia to build our app and then put aforementioned rendering in part of the screen or as a background and allow it to consume touch input.

Thank you,
Nemanja

**Comment by wieslawsoltes** [comment]
You can check https://github.com/TheKeyblader/Unilonia

**Reply by NemanjaBozovic-TomTom** [reply]
Seems like it's abandoned, no updates after 2020.

**Reply by maxkatz6** [reply]
You need to implement platform specific interfaces to get it work with Unity. That project from above can be used as a reference.
There weren't any fundamental changes in Avalonia since 2020, so it should be possible to update that code.

**Reply by NemanjaBozovic-TomTom** [reply]
Okay, thanks :)

**Reply by 0use-TE** [reply]
@NemanjaBozovic-TomTom Compile Unity into WASM, then use WebView for rendering, and communicate via WebSocket. How about this idea?
